### PR TITLE
Update WC_Admin_Duplicate_Product to use CRUD.

### LINKED ISSUE #12914: Duplicate product code needs to use CRUD
Can be simplified greatly becasue it should just include cloning, setting ID to 0 and resaving.

### PR DESCRIPTION
**Pull request by emdashcodes**
Closes #12914.

Yay for red PRs!

### CODE REVIEW

File: includes/admin/class-wc-admin-duplicate-product.php [hunk 1]
```diff
@@ -79,143 +79,49 @@ public function dupe_button() {
...
-		// Copy the meta information
-		$this->duplicate_post_meta( $post->ID, $new_post_id );
-
-		// Copy the children (variations)
-		$exclude = apply_filters( 'woocommerce_duplicate_product_exclude_children', false );
-
-		if ( ! $exclude && ( $children_products = get_children( 'post_parent=' . $post->ID . '&post_type=product_variation' ) ) ) {
-			foreach ( $children_products as $child ) {
-				$this->duplicate_product( $this->get_product_to_duplicate( $child->ID ), $new_post_id, $child->post_status );
+		if ( $product->is_type( 'variable' ) ||  $product->is_type( 'grouped' ) ) {
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-duplicate-product.php]: Extra whitespace after the `||` here. There is a linter for Atom which should catch this if you're not using it :)

File: includes/admin/class-wc-admin-duplicate-product.php [hunk 2]
```diff
@@ -79,143 +79,49 @@ public function dupe_button() {
...
+				$child_duplicate->set_id( 0 );
+				$child_duplicate->save();
 			}
 		}
 
-		// Clear cache
-		clean_post_cache( $new_post_id );
+		// Hook rename to match other woocommerce_product_* hooks, and to move away from depending on a response from the wp_posts table.
+		// New hook returns new id and old id.
+		do_action( 'woocommerce_product_duplicate', $duplicate->get_id(), $product_id );
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-duplicate-product.php]: Since this is a new hook, should we pass objects rather than ids?

**Review by mikejolley — CHANGES_REQUESTED** [review]
Added some feedback. CRUD joy to use here right :)

One thing missing for products and variations; make sure the SKU stats unique. I added a `wc_product_force_unique_sku` function a few weeks ago which should help with that.

**Comment by justinshreve** [comment]
Feedback handled.

@mikejolley What linter are you using? I was using linter-php but it doesn't catch spacing errors. I tried linter-phpcs just now with WordPress rules, but it also didn't catch the extra space.

**Comment by mikejolley** [comment]
@justinshreve these two https://dl.dropboxusercontent.com/s/1rlcd5yyx2rjonh/2017-01-25%20at%2019.58.png

**Pull request merged**